ports/esp32/network_common.c

```diff
@@ -154,6 +155,148 @@ static mp_obj_t esp_ifconfig(size_t n_args, const mp_obj_t *args) {
+        if (n_args != 0) {
+            mp_raise_TypeError(MP_ERROR_TEXT("can't specify pos and kw args"));
+        }
+        int touched_ip_info = 0;
+        int touched_dns_info = 0;
+        for (size_t i = 0; i < kwargs->alloc; ++i) {
+            if (MP_MAP_SLOT_IS_FILLED(kwargs, i)) {
+                mp_map_elem_t *e = &kwargs->table[i];
+                switch (mp_obj_str_get_qstr(e->key)) {
+                    case MP_QSTR_dhcp4: {
+                        esp_netif_dhcp_status_t status;
+                        if (self->if_id == ESP_IF_WIFI_STA || self->if_id == ESP_IF_ETH) {
+                            esp_exceptions(esp_netif_dhcpc_get_status(self->netif, &status));
+                            if (mp_obj_is_true(e->value) && status != ESP_NETIF_DHCP_STARTED) {
+                                esp_exceptions(esp_netif_dhcpc_start(self->netif));
+                            } else if (!mp_obj_is_true(e->value) && status == ESP_NETIF_DHCP_STARTED) {
+                                esp_exceptions(esp_netif_dhcpc_stop(self->netif));
+                            }
+                        } else {
+                            mp_raise_ValueError(MP_ERROR_TEXT("unexpected key"));
+                            break;
+                        }
+                        break;
+                    }
+                    case MP_QSTR_addr4: {
+                        int prefix_bits = 32;
+                        if (e->value != mp_const_none && mp_obj_is_str(e->value)) {
+                            size_t addr_len;
+                            const char *input_str = mp_obj_str_get_data(e->value, &addr_len);
+                            char plain_ip[IPADDR_STRLEN_MAX];
+                            char *split = strchr(input_str, '/');
+                            const char *addr_str = input_str;
+                            if (split) {
+                                int to_copy = sizeof(plain_ip) - 1;
+                                if (split - addr_str < to_copy) {
+                                    to_copy = split - addr_str;
+                                }
+                                memcpy(plain_ip, addr_str, to_copy);
+                                mp_obj_t prefix_obj = mp_parse_num_integer(split + 1, strlen(split + 1), 10, NULL);
+                                prefix_bits = mp_obj_get_int(prefix_obj);
+                            }
+                            uint32_t mask = -(1u << (32 - prefix_bits));
+                            uint32_t *m = (uint32_t *)&info.netmask;
+                            *m = esp_netif_htonl(mask);
+                            // TODO: netutils code does not allow inputing a C-string, but requires a mp_string
+                            // This makes parsing CIDR-Notation a bit more complex.
+                            // netutils_parse_ipv4_addr(addr_str, (void *)&info.ip, NETUTILS_BIG);
+                        } else if (e->value != mp_const_none) {
```

Comment:
What I did for NINAW10 is to create a temporary string object with to_copy set to addr_len as default. So if the / is missing, just the address is taken.
```
                            mp_obj_t ip_addr_obj = mp_obj_new_str(addr_str, to_copy);
                            netutils_parse_ipv4_addr(ip_addr_obj, ifconfig.ip_addr, NETUTILS_BIG);
```
The other option would be to change `netutils_parse_ipv4_addr()` to stop parsing a number at '/' like it does for '.'. Then e->value can be used.

Reply:
So I changed `netutils_parse_ipv4_addr()` in netutils.c to stop the IP scanning at a '/' by adding the lines
```
    char *split = strchr(addr_str, '/');
    const char *s_top = split ? split : addr_str + addr_len;
```
Then, no interim string object is needed and e->value can be used as input to `netutils_parse_ipv4_addr()`.

Reply:
I changed `netutils_parse_ipv4_addr()` again to stop at any character not being a digit or '.'. That generally improves the quality of parsing and has the side effect, that anything up from a '/' is ignored, which we use here.